ORSSamples/Program.cs

```diff
@@ -0,0 +1,192 @@
+            blobUpdater(sourceContainerClient, destContainerClient, blobName, blobContent2);          
+            archiveContainerFiles(destContainerClient, destConnectionString, destContainerName);
+            // archiveContainerFilesUsingBatch(destServiceClient, destContainerClient);
+        }
+
+        /*
+         * Uploads 1000 blobs, then checks status of replication in interval of 1 min. Outputs status as a percentage of blobs completed replication
+         */
+        public static void multipleBlobUpdater(BlobContainerClient sourceContainerClient, BlobContainerClient destContainerClient, String blobName, String blobContent, int numberBlobs)
+        {
+            // Uploading blobs
+            Console.WriteLine("Demonstrating tracking percentage of blobs completed");
+            List<string> blobNames = new List<string>();
+            Console.WriteLine("Uploading " + numberBlobs + " blobs");
+            for (int i = 0; i < numberBlobs; i++)
+            {
+                String newBlobName = i + blobName;
+                BlobClient sourceBlob = sourceContainerClient.GetBlobClient(newBlobName);
+                Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(blobContent));
+                sourceBlob.Upload(stream, true);
+                blobNames.Add(newBlobName);
+            }
+
+            // Check if replication in dest container is finished in interval of 5 min
+            Console.WriteLine("Checking to see if replication finished");
+            while (blobNames.Count > 0)
+            {
+                Thread.Sleep(60000);
+                List<string> repIncomplete = new List<string>();
+                foreach (String currBlobName in blobNames)
+                {
+                    bool status = checkReplicationStatus(sourceContainerClient, currBlobName);
+                    if (!status)
+                    {
+                        repIncomplete.Add(currBlobName);
+                    }
+                }
+                blobNames = repIncomplete;
+                Console.WriteLine("Percentage completed: " + ((numberBlobs - blobNames.Count) / numberBlobs * 100 + "%"));
+                Console.WriteLine((numberBlobs - blobNames.Count) + " out of " + numberBlobs + " blobs have completed");
+            }
+        }
+
+        /*
+         * Checks replication status of given blob in given container
+         */
+        public static bool checkReplicationStatus(BlobContainerClient sourceContainerClient, String blobName)
+        {
+            BlobClient sourceBlob = sourceContainerClient.GetBlobClient(blobName);
+            Response<BlobProperties> source_response = sourceBlob.GetProperties();
+            IList<ObjectReplicationPolicy> policyList = source_response.Value.ObjectReplicationSourceProperties;
+
+            // If policyList is null, then replication still in progress
+            if (policyList != null)
```

Comment:
I'm surprised that it's in the case of the policy list not existing. I don't think the blob should exist in general if the replication is not complete (e.g. we should have gotten a 404 BlobNotFound after the GetProperties call, but I could be wrong about this since it seems to be working in your case). Sometimes customers can set their rules to only copy say the JPG or TXT files but not the other files. So we could have blobs in the destination container not related to the replication showing up and then we could. But on the other hand this is a demonstration, so maybe when the customer is setting up their storage accounts and ORS policies and rules to make them general, and have empty containers as well.

Reply:
Since I'm checking the replication status of the blobs in the source container, I thought that the blob has to exist since I uploaded them as part of the beginning of my program?

Reply:
ohh you're totally right. For some reason I thought this was getting the properties on the destination blob, my bad!